model/patch_lifecycle.go

```diff
@@ -430,3 +434,104 @@ func AbortPatchesWithGithubPatchData(createdBefore time.Time, owner, repo string
+	}
+	cq, err := commitqueue.FindOneId(projectID)
+	if err != nil {
+		return 0, errors.Wrapf(err, "error finding commit queue '%s'", projectID)
+	}
+	if cq == nil {
+		return 0, errors.New("commit queue '%s' not found")
+	}
+
+	if patchType == commitqueue.PRPatchType {
+		res, err := restartPRItems(patches, cq)
+		return res, errors.Wrapf(err, "error re-adding %s patch items to queue", commitqueue.PRPatchType)
+	}
+	if patchType == commitqueue.CLIPatchType {
+		res, err := restartCLIItems(patches, cq)
+		return res, errors.Wrapf(err, "error re-adding %s patch items to queue", commitqueue.CLIPatchType)
+	}
+	return 0, errors.Errorf("patch type '%s' is invalid", patchType)
+}
+
+func restartPRItems(patches []patch.Patch, cq *commitqueue.CommitQueue) (int, error) {
+	numRestarted := 0
+	for _, p := range patches {
+		// reconstruct commit queue item from patch
+		modules := []commitqueue.Module{}
+		for _, modulePatch := range p.Patches {
+			module := commitqueue.Module{
+				Module: modulePatch.ModuleName,
+				Issue:  modulePatch.PatchSet.Patch,
+			}
+			modules = append(modules, module)
+		}
+		item := commitqueue.CommitQueueItem{
+			Issue:   strconv.Itoa(p.GithubPatchData.PRNumber),
+			Modules: modules,
+		}
+		if _, err := cq.Enqueue(item); err != nil {
+			return numRestarted, errors.Wrapf(err, "error enqueueing patch item")
+		}
+		numRestarted++
+		// TODO: might want to do some logging things also
+	}
+	return numRestarted, nil
+}
+
+func restartCLIItems(patches []patch.Patch, cq *commitqueue.CommitQueue) (int, error) {
+	// create new patch
+	numRestarted := 0
+	for _, p := range patches {
+		u, err := user.FindOne(user.ById(p.Author))
+		if err != nil {
+			return numRestarted, errors.Wrap(err, "error finding user for patch")
+		}
+		if u == nil {
+			return numRestarted, errors.Wrap(err, "user for patch not found")
+		}
+		patchNumber, err := u.IncPatchNumber()
+		if err != nil {
+			return numRestarted, errors.Wrap(err, "error computing patch num")
+		}
+		newPatch := patch.Patch{
+			Id:              mgobson.NewObjectId(),
+			Project:         p.Project,
+			Author:          p.Author,
+			Githash:         p.Githash,
+			CreateTime:      time.Now(),
+			Status:          evergreen.PatchCreated,
+			Description:     p.Description,
+			GithubPatchData: p.GithubPatchData,
+			Tasks:           p.Tasks,
+			VariantsTasks:   p.VariantsTasks,
+			BuildVariants:   p.BuildVariants,
+			Alias:           p.Alias,
+			Patches:         p.Patches,
+			PatchNumber:     patchNumber,
+		}
+
+		if err = newPatch.Insert(); err != nil {
+			return numRestarted, errors.Wrap(err, "error creating new patch")
+		}
+		modules := []commitqueue.Module{}
+		for _, modulePatch := range newPatch.Patches {
+			modules = append(modules, commitqueue.Module{Module: modulePatch.ModuleName})
+			// TODO: should this module also have Issue?
+		}
+		item := commitqueue.CommitQueueItem{
+			Issue:   newPatch.Id.Hex(),
+			Modules: modules,
```

Comment:
I'm not sure it's necessary to include the modules in the item for CLI items
https://github.com/evergreen-ci/evergreen/blob/869d83e64571178a71cea907b3d505a6519a46f0/rest/route/commit_queue.go#L161